core/src/main/java/com/linecorp/armeria/server/ServiceRequestContextBuilder.java

```diff
@@ -228,20 +228,23 @@ public ServiceRequestContext build() {
         final InetAddress clientAddress = server.config().clientAddressMapper().apply(proxiedAddresses)
                                                 .getAddress();
 
+        EventLoop eventLoop = eventLoop();
+        if (eventLoop == null) {
+            eventLoop = CommonPools.workerGroup().next();
+        }
+
         final CancellationScheduler requestCancellationScheduler;
         if (timedOut()) {
             requestCancellationScheduler = CancellationScheduler.finished(true);
         } else {
             requestCancellationScheduler = CancellationScheduler.ofServer(0);
-            requestCancellationScheduler.initAndStart(eventLoop(), noopCancellationTask);
+            requestCancellationScheduler.initAndStart(eventLoop, noopCancellationTask);
         }
 
-        final EventLoop serviceWorkerGroup = eventLoop();
-
         // Build the context with the properties set by a user and the fake objects.
         final Channel ch = fakeChannel();
```

Comment:
Should we use the `EventLoop` acquired before?
```suggestion
        final Channel ch = fakeChannel(eventLoop);
```

Reply:
That seems more suitable for testing! Thank you!